src/platform/openiotsdk/OpenIoTSDKArchUtils.c

```diff
@@ -46,31 +46,16 @@ uint64_t GetTick(void)
 /* Time to kernel ticks */
 uint32_t sec2tick(uint32_t sec)
 {
-    if (sec == 0U)
-    {
-        return (osWaitForever);
-    }
-
     return (sec * osKernelGetTickFreq());
 }
 
 uint32_t ms2tick(uint32_t ms)
 {
-    if (ms == 0U)
-    {
-        return (osWaitForever);
-    }
-
     return (uint32_t)(((uint64_t) ms * (uint64_t) osKernelGetTickFreq()) / 1000U);
 }
 
 uint32_t us2tick(uint32_t usec)
 {
-    if (usec == 0U)
-    {
-        return osWaitForever;
-    }
```

Comment:
Do you know why we were returning osWaitForever when the input was 0 ? 
The change looks correct with the usage made of the tick conversion functions.

Reply:
I don't know but now it causes an issue in the event loop.